app/notifications/process_notifications.py

```diff
@@ -44,7 +49,34 @@
 
 
 def create_content_for_notification(template, personalisation):
-    template_object = template._as_utils_template_with_personalisation(personalisation)
+    if template.template_type == EMAIL_TYPE:
+        template_object = PlainTextEmailTemplate(
+            {
```

Comment:
Wonder if instead of explicitely having the logic here for what fields we expect to pass to the different template types we should just pass in all the keys for the template and then the class can decide which it wants to use?

Something like

```
PlainTextEmailTemplate(
    template.serialize()
    personalisation
)
```

Reply:
Reason for doing it this way is deliberate, see 4d041f7